go/vt/binlog/binlogplayer/binlog_player.go

```diff
@@ -659,12 +661,12 @@ func GenerateUpdateRowsCopied(uid uint32, rowsCopied int64) string {
 	return fmt.Sprintf("update _vt.vreplication set rows_copied=%v where id=%v", rowsCopied, uid)
 }
 
-// GenerateUpdateTime returns a statement to update time_updated in the _vt.vreplication table.
-func GenerateUpdateTime(uid uint32, timeUpdated int64) (string, error) {
+// GenerateUpdateHeartbeat returns a statement to record the latest heartbeat in the _vt.vreplication table.
+func GenerateUpdateHeartbeat(uid uint32, timeUpdated int64) (string, error) {
 	if timeUpdated == 0 {
 		return "", fmt.Errorf("timeUpdated cannot be zero")
 	}
-	return fmt.Sprintf("update _vt.vreplication set time_updated=%v where id=%v", timeUpdated, uid), nil
+	return fmt.Sprintf("update _vt.vreplication set time_updated=%v, time_heartbeat=%v where id=%v", timeUpdated, timeUpdated, uid), nil
```

Comment:
Is the `time_heartbeat` ever different than `time_updated` in the current code? I didn't immediately see where they differ. Not sure if I'm missing something or perhaps we are allowing for future behavior.

Reply:
When a transaction is seen and the `pos` column is updated with the recent gtid we also update `time_updated` (and not `time_heartbeat`)